packages/compass/src/app/components/workspace.tsx

```diff
@@ -87,31 +92,38 @@ export default function Workspace({
         >
           <WorkspacesPlugin
             initialWorkspaceTabs={[{ type: 'My Queries' }]}
+            singleConnectionConnectionInfo={singleConnectionConnectionInfo}
             onActiveWorkspaceTabChange={onActiveWorkspaceTabChange}
-            renderSidebar={() => {
-              return (
-                <CompassSidebarPlugin
-                  initialConnectionInfo={connectionInfo ?? undefined}
-                />
-              );
-            }}
+            renderSidebar={() => (
+              <CompassSidebarPlugin
+                singleConnectionConnectionInfo={singleConnectionConnectionInfo}
+              />
+            )}
             renderModals={() => {
-              return (
-                <>
+              return multiConnectionsEnabled ? (
+                <RenameCollectionPlugin></RenameCollectionPlugin>
+              ) : (
+                <ConnectionInfoProvider
+                  connectionInfoId={singleConnectionConnectionInfo?.id}
+                >
                   <ImportPlugin></ImportPlugin>
                   <ExportPlugin></ExportPlugin>
                   <CreateViewPlugin></CreateViewPlugin>
                   <CreateNamespacePlugin></CreateNamespacePlugin>
                   <DropNamespacePlugin></DropNamespacePlugin>
                   <RenameCollectionPlugin></RenameCollectionPlugin>
-                </>
+                </ConnectionInfoProvider>
               );
             }}
           ></WorkspacesPlugin>
         </CollectionTabsProvider>
       </WorkspacesProvider>
       <div className={shellContainerStyles}>
-        <CompassShellPlugin />
+        <ConnectionInfoProvider
+          connectionInfoId={singleConnectionConnectionInfo?.id}
+        >
+          <CompassShellPlugin />
+        </ConnectionInfoProvider>
       </div>
     </div>
   );
```

Comment:
Do we need similar changes in compass-web?

Reply:
I think compass-web injects it's own `ConnectionInfoProvider` so I think this should still work, but it's worth to test.

Reply:
So my note above was exactly about this - I wanted to confirm with Sergey if there is any relation between what we're doing in our sandbox and what will be done in mms. Thankfully there is not much. The keypoint is that mms will always know cluster name and will use that as connectionId to be resolved alongside the ConnectionInfo so I can safely rely on that passed info.

Reply:
Confirmed and fixed the issues with CompassWeb. The render tree for CompassWeb didn't change much as we're doing the single connection flow there.